src/config.c

```diff
@@ -775,15 +780,18 @@ void configSetCommand(client *c) {
         if (!apply_fns[i](&errstr)) {
             serverLog(LL_WARNING, "Failed applying new configuration. Possibly related to new %s setting. Restoring previous settings.", set_configs[config_map_fns[i]]->name);
             restoreBackupConfig(set_configs, old_values, config_count, apply_fns);
+            err_arg_name = set_configs[config_map_fns[i]]->name;
             goto err;
         }
     }
     addReply(c,shared.ok);
     goto end;
 
 err:
-    if (errstr) {
-        addReplyErrorFormat(c,"Config set failed - %s", errstr);
+    if (invalid_arg_name) {
+        addReplyErrorFormat(c,"Unknown option or number of arguments for CONFIG SET - '%s'", invalid_arg_name);
+    } else if (errstr) {
+        addReplyErrorFormat(c,"argument '%s' for CONFIG SET failed - %s", err_arg_name, errstr);
```

Comment:
As mentioned before, we can't be sure which argument (or combination of arguments) caused the "apply" failure. We can assume it's the first once, but I think this needs to be reflected in the error message:
```suggestion
        addReplyErrorFormat(c,"CONFIG SET failed (possibly related to argument '%s') - %s", err_arg_name, errstr);
```

Reply:
Yes, I agree with you. I check the codes and when the first error argument is checked, then Config set command is stopped, and restore all previous configs and popup the error message.

Reply:
> As mentioned before, we can't be sure which argument (or combination of arguments) caused the "apply" failure. We can assume it's the first once, but I think this needs to be reflected in the error message:

Address your comments, please take a look, thanks